third_party/move/move-compiler-v2/src/lib.rs

```diff
@@ -116,9 +124,84 @@ where
     run_bytecode_verifier(&annotated_units, &mut env);
     check_errors(&env, error_writer, "bytecode verification errors")?;
 
+    // Finally mark this model to be generated by v2
+    env.set_compiler_v2(true);
+
     Ok((env, annotated_units))
 }
 
+/// Run move compiler and print errors to given writer for the purpose of analysis, like
+/// e.g. the Move prover. After successful compilation attaches the generated bytecode
+/// to the model.
+pub fn run_move_compiler_for_analysis(
+    error_writer: &mut impl WriteColor,
+    mut options: Options,
+) -> anyhow::Result<GlobalEnv> {
+    options.whole_program = true; // will set `treat_everything_as_target`
```

Comment:
"Everything as target" is probably not is wanted in the general analysis case.  Our various compiler warning checks need to know the difference between "warn about this code" and "don't warn about this code".  Currently they are using "target" to decide whether to warn about code, but for certain targeted analyses you would want to do different things for "real target" and "target because it's a dependency".  That may mean we just need a new flag for "generate warnings/errors about this code".

Reply:
You usually expect that the prover is run an code which already passes type checks. But if not, we can't really differentiate whether the type check error is in the set targets or not, so we better produce errors in the dependencies as well. I think a really sustainable solution for the problem is that the package system somehow tracks this for us and/or we have some on-disk storage of compiled packages. Keeping for now like this until we have a better solution, perhaps after package system refactoring.